Application/ControlMsrE2/ControlMsrE2UI.c

```diff
@@ -0,0 +1,241 @@
+/** @file
+  Control MSR 0xE2 UI Input/Output in the widest sense
+
+Copyright (c) 2020, Brumbaer. All rights reserved.<BR>
+This program and the accompanying materials
+are licensed and made available under the terms and conditions of the BSD License
+which accompanies this distribution.  The full text of the license may be found at
+http://opensource.org/licenses/bsd-license.php
+
+THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+#include "ControlMsrE2.h"
+
+UINTN           Flags;
+
+/*
+  Read up to Length -1 Characters from keyboard.
+  CR will exit
+  LF will exit
+  Del key is supported
+  Esc any input will be cleared. If there isn't any ReadLine will exít
+  When length - 1 characters are entered Readline will exit automatically.
+*/
+UINT32
+ReadLine (
+  OUT CHAR16   *Buffer,
+  IN  UINT32   Length
+  )
+{
+  EFI_STATUS     Status;
+  UINTN          EventIndex;
+  EFI_INPUT_KEY  Key;
+  UINT32         Index;
+  UINT32         Pos;
+  INT32          StartRow;
+  INT32          StartColumn;
+
+  Pos = 0;
+
+  STATIC CHAR16  Output[] = L"A";
```

Comment:
Why is this STATIC? Why is its content "A"? If I'm not blind, this can never be read (i.e. is overwritten beforehand).

Reply:
I guess he want to get "A" "\0" here for further typewriter-like output and "A" is just a placeholder. Sounds like a imitation of https://github.com/acidanthera/OpenCorePkg/blob/master/Application/KeyTester/KeyTester.c#L34.